src/compiler/checker.ts

```diff
@@ -21115,20 +21113,27 @@ namespace ts {
                     return undefined;
                 }
 
-                const text = getTextOfPropertyName(name);
-                const type = isTypeAny(objectLiteralType)
-                    ? objectLiteralType
-                    : getTypeOfPropertyOfType(objectLiteralType, text) ||
-                    isNumericLiteralName(text) && getIndexTypeOfType(objectLiteralType, IndexKind.Number) ||
-                    getIndexTypeOfType(objectLiteralType, IndexKind.String);
-                if (type) {
-                    if (property.kind === SyntaxKind.ShorthandPropertyAssignment) {
-                        return checkDestructuringAssignment(property, type);
-                    }
-                    else {
-                        // non-shorthand property assignments should always have initializers
-                        return checkDestructuringAssignment(property.initializer, type);
+                let type: Type | undefined;
+                if (isTypeAny(objectLiteralType)) {
+                    type = objectLiteralType;
+                }
+                else {
+                    const text = getTextOfPropertyName(name);
+                    if (text) { // TODO: GH#26379
+                        const prop = getPropertyOfType(objectLiteralType, text);
+                        if (prop) {
+                            markPropertyAsReferenced(prop, property, rightIsThis);
+                            checkPropertyAccessibility(property, /*isSuper*/ false, objectLiteralType, prop);

```

Comment:
Should you be getting type only if propertyAccessibility doesn't have issues?

Reply:
It looks like we normally just call this for the diagnostic and don't change the type, e.g.:
```ts
class C { private x!: number; }
const x = new C().x; // Error, but doesn't affect type
x; // number
```